services/ext/service.go

```diff
@@ -645,24 +644,51 @@ func (s *Service) FetchCommunityInfo(communityID string) (*thirdparty.CommunityI
 	return communityToInfo(community), nil
 }
 
+func (s *Service) FetchCommunityInfoForCollectibles(communityID string, ids []thirdparty.CollectibleUniqueID) (*thirdparty.CommunityInfo, error) {
+	community, err := s.messenger.FindCommunityInfoFromDB(communityID)
+	if err != nil && err != communities.ErrOrgNotFound {
+		return nil, err
+	}
+
+	if community == nil {
+		return s.FetchCommunityInfo(communityID)
+	}
+
+	hasAllMetadata := true
+	for _, id := range ids {
+		tokenMetadata, err := s.fetchCommunityCollectibleMetadata(community, id.ContractID)
```

Comment:
premature optimization incoming: we only need to look once per contract ID, if the user owns 1000 collectibles from the same contract which is already in the DB, this is gonna do 999 unnecessary lookups